src/main/java/uk/gov/hmcts/reform/sscs/controller/CcdMideventCallbackController.java

```diff
@@ -142,6 +149,55 @@ public ResponseEntity<PreSubmitCallbackResponse<SscsCaseData>> ccdMidEventPrevie
         return ok(adjournCasePreviewService.preview(callback, DocumentType.DRAFT_ADJOURNMENT_NOTICE, userAuthorisation, false));
     }
 
+    @PostMapping(path = "/ccdMidEventAdjournCaseDirectionDueDate", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<PreSubmitCallbackResponse<SscsCaseData>> ccdMidEventAdjournDirectionDueDate(
+            @RequestHeader(SERVICE_AUTHORISATION_HEADER) String serviceAuthHeader,
+            @RequestBody String message) {
+        Callback<SscsCaseData> callback = deserializer.deserialize(message);
+        log.info("About to start ccdMidEventAdjournDirectionDueDate callback `{}` received for Case ID `{}`", callback.getEvent(),
+                callback.getCaseDetails().getId());
+        authorisationService.authorise(serviceAuthHeader);
+        SscsCaseData caseData = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<SscsCaseData> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(caseData);
+        adjournCaseMidEventValidationService.validateSscsCaseDataConstraints(caseData, preSubmitCallbackResponse);
+        try {
```

Comment:
since we're passing the preSubmitCallbackResponse to the other method anyway, we might as well have another method in there for the directionsDueDate and move the try-catch there
I think one way of doing it without passing the preSubmitCallbackResponse around is for the validate methods to return an error list, and then we can just do 
`
preSubmitCallbackResponse.addErrors(validateDirectionsDueDate(sscsCaseData))
`